ipset.go

```diff
@@ -237,12 +277,18 @@ func discardf(format string, args ...interface{}) {}
 var debugf = discardf
 
 // IPSet returns an immutable IPSet representing the current state of
-// s. The builder remains usable after calling IPSet.
-func (s *IPSetBuilder) IPSet() *IPSet {
+// s, along with any errors accumulated during the construction of s.
```

Comment:
I'm glad this is documented but I also feel like APIs that can return a partially constructed object along with errors are kind of weird to deal with. Are there legitimate reasons why one might want this capability?

Reply:
It was so people could ignore this error in the common case, as none of the errors really matter much usually.

But @danderson, I'd document it more explicitly. It does say "along with" but the norm in Go APIs is so strongly "one or the other" that things that return two non-zero values really need to be yelled.

It probably doesn't need to be as verbose as, say, https://golang.org/pkg/io/#Reader but a bit closer.